controllers/state_of_the_world.go

```diff
@@ -59,6 +64,72 @@ func NewPolicyMachineryController(manager ctrlruntime.Manager, client *dynamic.D
 		controller.WithReconcile(buildReconciler(client)),
 	}
 
+	ok, err := kuadrantgatewayapi.IsGatewayAPIInstalled(manager.GetRESTMapper())
+	if err != nil || !ok {
+		logger.Info("gateway api is not installed, skipping watches and reconcilers", "err", err)
+	} else {
+		controllerOpts = append(controllerOpts,
+			controller.WithRunnable("gatewayclass watcher", controller.Watch(&gwapiv1.GatewayClass{}, controller.GatewayClassesResource, metav1.NamespaceAll)),
+			controller.WithRunnable("gateway watcher", controller.Watch(&gwapiv1.Gateway{}, controller.GatewaysResource, metav1.NamespaceAll)),
+			controller.WithRunnable("httproute watcher", controller.Watch(&gwapiv1.HTTPRoute{}, controller.HTTPRoutesResource, metav1.NamespaceAll)),
+		)
+	}
+
+	ok, err = envoygateway.IsEnvoyGatewayInstalled(manager.GetRESTMapper())
+	if err != nil || !ok {
+		logger.Info("envoygateway is not installed, skipping related watches and reconcilers", "err", err)
+	} else {
+		controllerOpts = append(controllerOpts,
+			controller.WithRunnable("envoypatchpolicy watcher", controller.Watch(&egv1alpha1.EnvoyPatchPolicy{}, envoygateway.EnvoyPatchPoliciesResource, metav1.NamespaceAll)),
+			controller.WithRunnable("envoyextensionpolicy watcher", controller.Watch(&egv1alpha1.EnvoyExtensionPolicy{}, envoygateway.EnvoyExtensionPoliciesResource, metav1.NamespaceAll)),
+			controller.WithRunnable("envoysecuritypolicy watcher", controller.Watch(&egv1alpha1.SecurityPolicy{}, envoygateway.SecurityPoliciesResource, metav1.NamespaceAll)),
+			controller.WithObjectKinds(
+				envoygateway.EnvoyPatchPolicyGroupKind,
+				envoygateway.EnvoyExtensionPolicyGroupKind,
+				envoygateway.SecurityPolicyGroupKind,
+			),
+			// TODO: add object links
+		)
+		// TODO: add specific tasks to workflow
```

Comment:
Future TODOs that would be completed by other issues as functionality is migrated to the workflow